query-engine/connectors/sql-query-connector/src/row.rs

```diff
@@ -92,12 +92,13 @@ impl ToSqlRow for ResultRow {
 
         for (i, p_value) in self.into_iter().enumerate().take(row_width) {
             let pv = match (meta[i].identifier(), meta[i].arity()) {
-                (type_identifier, FieldArity::List) => match p_value {
+                (type_identifier, FieldArity::List) => match p_value.typed {
                     value if value.is_null() => Ok(PrismaValue::List(Vec::new())),
-                    Value::Array(None) => Ok(PrismaValue::List(Vec::new())),
-                    Value::Array(Some(l)) => l
+                    ValueType::Array(None) => Ok(PrismaValue::List(Vec::new())),
+                    // TODO: avoid cloning in here.
+                    ValueType::Array(Some(l)) => l
                         .into_iter()
-                        .map(|p_value| row_value_to_prisma_value(p_value, meta[i]))
+                        .map(|val| row_value_to_prisma_value(val.clone(), meta[i]))
```

Comment:
Why do we need to clone here? This compiles as well:
```suggestion
                        .map(|val| row_value_to_prisma_value(val, meta[i]))
```

Reply:
We don't this was a left over from an intermediate state. I believe. Fixing.